wolfcrypt/src/port/iotsafe/iotsafe.c

```diff
@@ -775,14 +848,13 @@ static int iotsafe_sign_hash(byte *privkey_idx, uint16_t id_size,
 
         ret = expect_csim_response(csim_cmd, (word32)XSTRLEN(csim_cmd), &resp);
         if (ret >= 0) {
-            byte sig_hdr[3];
-            if (hex_to_bytes(resp, sig_hdr, 3) < 0) {
+            byte sig_hdr[2];
+            if (hex_to_bytes(resp, sig_hdr, 2) < 0) {
                ret = BAD_FUNC_ARG;
             } else if ((sig_hdr[0] == IOTSAFE_TAG_SIGNATURE_FIELD) &&
-                       (sig_hdr[1] == 0) &&
-                       (sig_hdr[2] == 2 * IOTSAFE_ECC_KSIZE)) {
-                XSTRNCPY(R, resp + 6, IOTSAFE_ECC_KSIZE * 2);
-                XSTRNCPY(S, resp + 6 + IOTSAFE_ECC_KSIZE * 2,
+                       (sig_hdr[1] ==  2 * IOTSAFE_ECC_KSIZE)) {
+                XSTRNCPY(R, resp + 4, IOTSAFE_ECC_KSIZE * 2);
+                XSTRNCPY(S, resp + 4 + IOTSAFE_ECC_KSIZE * 2,
```

Comment:
These changes in the `sign` operation break the parser for all other applets under test. The correct response format for the `Compute Signature` operation is the following:
- Tag `0x33` "Computed signature" with two-bytes len field (range is from 0x40 to 0x0100). [IoT.05 section 2.10.4.1]
- Signature data (R|S)

Please double check the format of the data returned by the applet (0x33 must have a 2 bytes length)

Reply:
note: an invalid signature is parsed back from the `compute signature` operation, when running these changes in the other applets. As a result, the other endpoint will fail to authenticate the iotsafe host.

Reply:
For this one GSMA indicates the range but they don't advise on the len byte (1 or 2) fixed byte. I will also check for this one.

Reply:
The GSMA spec is not clear enough. Our IoT Safe expert mention that for ECC the len is on 1 byte and on 2 bytes if you use RSA.
You do have a condition if it is ECC or RSA. technically when RSA will be supported then we can change inside the RSA condition the len to be encoded on 2 bytes. Otherwise for ECC as far as I know it is always 1 byte.  Maybe we should ask GSMA to be more specific here.

` if (sign_algo == IOTSAFE_SIGN_ECDSA) {...}
else {
ret = NOT_COMPILED_IN; /* RSA not yet supported */
}`

Reply:
if ECC it is always 1 byte, RSA 2 bytes because ECC is only 256bits while RSA is 2048 bits

Reply:
- RSA is not yet supported. Feel free to add the `if (sign_algo ==` check above.
- I agree that the IoT.05 spec is not 100% clear on tag 0x33  len field (in 2.10.4.1), but if the field may contain a 2-bytes length due to its range, then it should always be parsed as a 2 bytes field (there is no other way to predict the size of L in a TLV). Other len fields have implicit size, that depend on their range.
- In section 2.24.3.2, the same TLV '0x33' is specified as argument for the `verify` operation. Here the TLV is explicitly showing that the length is a 2 octets field, due to its range. 
- The same tag 0x33 is used in the code in `iotsafe_verify_hash`, where the TLV is used as parameter and added with its two bytes length.

My interpretation, together with the behavior implemented by two other applets, points in the direction of TLV 0x33 having 2Bytes length. I would suggest that you review the implementation of the applet you are using, accordingly. Unfortunately, I could not accept such a change in the iotsafe code because it breaks the `compute signature` operation for the other two demos.

I agree that this could be furtherly verified with GSMA if needed.

Reply:
Fantastic I will revert this change too. Thanks for pointing this out :)